hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/v2/components/tables/containersTable.tsx

```diff
@@ -0,0 +1,263 @@
+  },
+  {
+    title: 'Date Created',
+    dataIndex: 'CreationTime',
+    key: 'CreationTime',
+    render: (date: string) => getFormattedTime(date, 'lll')
+  },
+  {
+    title: 'Date Modified',
+    dataIndex: 'ModificationTime',
+    key: 'ModificationTime',
+    render: (date: string) => getFormattedTime(date, 'lll')
+  },
+  {
+    title: 'Path',
+    dataIndex: 'CompletePath',
+    key: 'path'
+  }
+];
+
+const ContainerTable: React.FC<ContainerTableProps> = ({
+  data,
+  loading,
+  selectedColumns,
+  expandedRow,
+  expandedRowSetter,
+  searchColumn = 'containerID',
+  searchTerm = ''
+}) => {
+
+  const cancelSignal = useRef<AbortController>();
+
+  function filterSelectedColumns() {
+    const columnKeys = selectedColumns.map((column) => column.value);
+    return COLUMNS.filter(
+      (column) => columnKeys.indexOf(column.key as string) >= 0
+    );
+  }
+
+  function loadRowData(containerID: number) {
+    const { request, controller } = AxiosGetHelper(
+      `/api/v1/containers/${containerID}/keys`,
+      cancelSignal.current
+    );
+    cancelSignal.current = controller;
+
+    request.then(response => {
+      const containerKeysResponse: ContainerKeysResponse = response.data;
+      expandedRowSetter({
+        ...expandedRow,
+        [containerID]: {
+          ...expandedRow[containerID],
+          loading: false,
+          dataSource: containerKeysResponse.keys,
+          totalCount: containerKeysResponse.totalCount
+        }
+      });
+    }).catch(error => {
+      expandedRowSetter({
+        ...expandedRow,
+        [containerID]: {
+          ...expandedRow[containerID],
+          loading: false
+        }
+      });
+      showDataFetchError((error as AxiosError).toString());
+    });
+  }
+
+  function getFilteredData(data: Container[]) {
+
+    return data?.filter(
+      (container: Container) => {
+        return (searchColumn === 'containerID')
+          ? container[searchColumn].toString().includes(searchTerm)
+          : container[searchColumn].includes(searchTerm)
+      }
+    ) ?? [];
+  }
+
+  function onRowExpandClick(expanded: boolean, record: Container) {
+    if (expanded) {
+      loadRowData(record.containerID);
+    }
+    else {
+      cancelSignal.current && cancelSignal.current.abort();
+    }
+  }
+
+  function expandedRowRender(record: Container) {
+    const containerId = record.containerID
+    const containerKeys: ExpandedRowState = expandedRow[containerId];
+    const dataSource = containerKeys?.dataSource?.map(record => ({
+      ...record,
+      uid: `${record.Volume}/${record.Bucket}/${record.Key}`
+    })) ?? [];
+    const paginationConfig: TablePaginationConfig = {
+      showTotal: (total: number, range) => `${range[0]}-${range[1]} of ${total} Keys`
+    }
+
+    return (
+      <Table
+        loading={containerKeys?.loading ?? true}
+        dataSource={dataSource}
+        columns={KEY_TABLE_COLUMNS}
+        pagination={paginationConfig}
+        rowKey='uid'
```

Comment:
I would prefer to use a function to derive the `rowKey` than creating a new attribute in `KeyResponse`

`rowKey={(record: KeyResponse) => `${record.Volume}/${record.Bucket}/${record.Key}`}

Reply:
You can refer to the `rowKey` in `buckets.tsx`

Reply:
I also noticed that the `rowKey` for `bucketsTable` is only 'volume`. It should be `{volume}_{bucket}`. Can fix here or raise a ticket.

Reply:
Addressed this in [98e177e](https://github.com/apache/ozone/pull/7267/commits/98e177e3600c573541aca51cbde5a3c6b434ed01) for both the bucketTable and containerTable